astacus/common/rohmustorage.py

```diff
@@ -185,23 +186,24 @@ def delete_json(self, name: str) -> None:
         key = os.path.join(self.json_key, name)
         self.storage.delete_key(key)
 
-    def download_json(self, name: str) -> Json:
+    @contextlib.contextmanager
+    def open_json_bytes(self, name: str) -> Iterator[bytearray]:
         key = os.path.join(self.json_key, name)
-        f = io.BytesIO()
-        self._download_key_to_file(key, f)
-        f.seek(0)
-        return json.load(f)
+        with tempfile.TemporaryFile(dir=self.config.temporary_directory) as temp_file:
+            self._download_key_to_file(key, temp_file)
+            temp_file.seek(0)
+            with mmap.mmap(temp_file.fileno(), 0, access=mmap.ACCESS_READ) as mapped_file:
```

Comment:
Considering we have ACCESS_READ here only, having the return type be `bytes` might make more sense actually.

Reply:
I think you meant that in a context where bytearray is a subclass of bytes, but since it's not the case, I don't think we can.

Reply:
This doesn't really apply anymore with things typed as mmap.